compiler/resolution/preFold.cpp

```diff
@@ -537,33 +537,56 @@ static Expr* preFoldPrimInitVarForManagerResource(CallExpr* call) {
 
         cc->getCalls(refCall, valueCall, constRefCall);
 
-        isAnyRefOverloadPresent = (constRefCall || refCall);
-        isConstnessUncertain = (constRefCall && refCall);
-        isCertainlyConstRef = (constRefCall && !refCall);
-
-        // Replace or remove the CC entirely if a ref option is present.
-        if (isAnyRefOverloadPresent && valueCall) {
-          if (refCall && constRefCall) {
-            auto swp = new ContextCallExpr();
+        // We note extra overloads in order from ref -> const ref -> value.
+        CallExpr* call = refCall ? refCall : constRefCall;
+        INT_ASSERT(call);
+
+        auto anyResolved = call->resolvedFunction();
+        INT_ASSERT(anyResolved);
+        INT_ASSERT(anyResolved->isMethod());
+
+        // TODO: What about if receiver is a primitive type?
+        auto at = anyResolved->getReceiverType();
+        INT_ASSERT(at);
+
+        USR_FATAL_CONT(lhs, "cannot determine resource storage due to "
```

Comment:
Can we say the name we can't determine the resource storage of?  E.g. "cannot determine resource storage of 'myResource' due to multiple return intents for..."?

Reply:
Good idea, how about "cannot determine storage of <name> due to multiple..."? I think we can actually omit the word 'resource' entirely then